examples/dynamic_example.jl

```diff
@@ -0,0 +1,234 @@
+#  Copyright 2015, Vincent Leclere, Francois Pacaud and Henri Gerard
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#############################################################################
+# Test SDDP with dam example
+# Source: Adrien Cassegrain
+#############################################################################
+
+#srand(2713)
+push!(LOAD_PATH, "../src")
+
+using StochDynamicProgramming, JuMP, Clp
+
+#Constant that the user have to define himself
+const SOLVER = ClpSolver()
+# const SOLVER = CplexSolver(CPX_PARAM_SIMDISPLAY=0)
+
+const N_STAGES = 5
+const N_SCENARIOS = 3
+
+const DIM_STATES = 2
+const DIM_CONTROLS = 2
+const DIM_ALEAS = 1
+
+
+
+#Constants that the user does not have to define
+# FINAL TIME:
+const TF = N_STAGES
+
+const T0 = 1
+const HORIZON = TF
+
+# Constants:
+const VOLUME_MAX = 100
+const VOLUME_MIN = 0
+
+const CONTROL_MAX = round(Int, .4/7. * VOLUME_MAX) + 1
+const CONTROL_MIN = 0
+
+const W_MAX = round(Int, .5/7. * VOLUME_MAX)
+const W_MIN = 0
+const DW = 1
+
+# Define aleas' space:
+const N_ALEAS = Int(round(Int, (W_MAX - W_MIN) / DW + 1))
+const ALEAS = linspace(W_MIN, W_MAX, N_ALEAS)
+
+const EPSILON = .05
+const MAX_ITER = 20
+
+alea_year = Array([7.0 7.0 8.0 3.0 1.0 1.0 3.0 4.0 3.0 2.0 6.0 5.0 2.0 6.0 4.0 7.0 3.0 4.0 1.0 1.0 6.0 2.0 2.0 8.0 3.0 7.0 3.0 1.0 4.0 2.0 4.0 1.0 3.0 2.0 8.0 1.0 5.0 5.0 2.0 1.0 6.0 7.0 5.0 1.0 7.0 7.0 7.0 4.0 3.0 2.0 8.0 7.0])
+
+const X0 = [50, 50]
+
+Ax=[]
+Au=[]
+Aw=[]
+
+Cx=[]
+Cu=[]
+Cw=[]
+
+function generate_random_dynamic()
+    for i=1:TF
+            push!(Ax, rand(DIM_STATES,DIM_STATES))
+            push!(Au, rand(DIM_STATES,DIM_CONTROLS))
+            push!(Aw, rand(DIM_STATES,DIM_ALEAS))
+    end
+end
+
+function generate_random_costs()
+    for i=1:TF
+            push!(Cx, rand(1,DIM_STATES))
+            push!(Cu, -1*rand(1,DIM_CONTROLS))
+            push!(Cw, rand(1,DIM_ALEAS))
+    end
+end
+
+generate_random_dynamic()
+generate_random_costs()
+
+# Define dynamic of the dam:
+function dynamic(t, x, u, w)
+    return  Ax[t]*x+Au[t]*u+Aw[t]*w
+end
+
+# Define cost corresponding to each timestep:
+function cost_t(t, x, u, w)
+    #if you want to compare the value, take the cost with only control
+    return (Cu[t]*u)[1,1]
+    #return (Cx[t]*x)[1,1]+(Cu[t]*u)[1,1]+(Cw[t]*w)[1,1]
```

Comment:
I think we need this line. The fact to have only the control let us compare SDDP and EF but if someone want a generic random cost, it can be good to let this line